### ISSUE TITLE
AAPS 2.7.0 rc4 Error 2049 Samsung Accessory SDK nog initialized.

### ISSUE THREAD

**Issue by FreDiabetics** [MilosKozak/AndroidAPS#2938]
I'm getting this error since I installed aaps 2.7.0 rc4 on my new Samsung Galaxy S20 
I installed aaps and after I restored my settings, aaps shows this at the bottm

Error 2049 Samsung Accessory SDK cannot be initialized. Device or build is not compatible

Also Sometimes my patched dex app gets readings but dont send it to aaps.
When this happens dex app is providing values with a  5 minutes delay. F.e new  112 in dex app on 8:00 but aaps recieves 112 at 08:05 and shows the last value is 10 minutes ago.... after a restart everythings working fine again.

I'm using aaps 2.7.0 rc4 with danaRS I did everything I always did when I'm setting up AAPS, automatic battery optimization  is deactivated and all the stuff I need to let AAPS do its work.

![Screenshot_20200827-184420_AndroidAPS.jpg](https://user-images.githubusercontent.com/45721291/91653176-85dff200-ea9e-11ea-80c9-cc7cd1b2c3ed.jpg)

**Comment by FreDiabetics** [comment]
Any thoughts on this issue @MilosKozak ?

**Comment by chrisbev** [comment]
This may be of little use but what I remember with my phone. Just changed to Samsung A41 but can't remember when I got the same (or similar SDK) error message. Don't think it stopped things but I have been changing builds rather a lot so don't remember which build I got the message or whether it was the first build with the phone? It appeared and I was alarmed but either reinstalled and it was OK or it didn't cause any apparent problem. I'd been switching between: stable release, RC and AT PR. Last couple of PR builds definitely not seen message, trying to recall but message probably only appeared once?

**Comment by FreDiabetics** [comment]
Yes for me too, I t appeared after importing my settings from my old Galaxy A80. I installed AAPS on my Galaxy S20 and imported settings, immediatly after that the message appeard. I uninstalled everything and started again. Installing, import settings and the error occured again. I t hasnt affected anything, so everything is working fine and I havent detetcted any issue after the message appeard.

I'll reset my phone in a few days to see if its different when I import settings from the same phone I uswed before, S20 on S20

**Comment by chrisbev** [comment]
Just tried back to RC2 and others and don't get message but import may be a clue as I just let new install do import. As a thought did you import from old or new format needing password?

**Comment by FreDiabetics** [comment]
I used 2.7.0 rc4 on old and new phone yes master password was needed but it was set on old and new phone. aaps on the new phone detected that I imported settings from an other phone but nothing about old or new format.

**Comment by chrisbev** [comment]
If file goes in root directory it is old format if goes in another folder //AAPS/preferences then new format and needs password. Just tried an import to 2.6.7B1 and no error message.

**Comment by FreDiabetics** [comment]
yes my files got exported in aaps/preferences so this is the new file format

**Comment by Atlas-Night-Out** [comment]
I've just got these issues in the last 3 days starting again with that error that keeps popping up on my mobile which is a Note10 + was on Android 10 now in the last 3 days gone to Android 11.
error: 2049 Samsung Accessory SDK cannot be initialized. Device or build not compatible
Can anyone help me with these issues? Do I need to update APS? I'm on version APS 2.8.2 Nightscout version: 14.2.2
